[DOCS-1166][DOCS-880][DOCS-1167][DOCS-1168] Update automation docs (#1084)
wandb/docs: content/guides/core/automations/create-automations/_index.md

{{% alert %}}
Looking for companion tutorials for automations? 
- [Learn to automatically triggers a Github Action for model evaluation and deployment](https://wandb.ai/wandb/wandb-model-cicd/reports/Model-CI-CD-with-W-B--Vmlldzo0OTcwNDQw).
- [Watch a video demonstrating automatically deploying a model to a Sagemaker endpoint](https://www.youtube.com/watch?v=s5CMj_w3DaQ).
- [Watch a video series introducing automations](https://youtube.com/playlist?list=PLD80i8An1OEGECFPgY-HPCNjXgGu-qGO6&feature=shared).
{{% /alert %}}

## Requirements
- A team admin can create and manage automations for the team's projects, as well as components of their automations, such as webhooks, secrets, or Slack instance integrations. Refer to [Team settings]({{< relref "/guides/models/app/settings-page/team-settings/" >}}).
- To create a registry automation, you must have access to the registry. Refer to [Configure Registry access]({{< relref "/guides/core/registry/configure_registry.md#registry-roles" >}}).
- To create a Slack automation, you must have permission to post to the Slack instance and channel you select.

> ## Create an automation

Reviewer: Can we structure this section a bit differently? It feels like we're trying cram some unrelated info into the sequence of steps, i.e authorization, what secrets are used for. 

If you are a team admin - you can optionally go to team settings and create:
secret
webhook or slack notification

If you are not, the above doesn't apply to you and you:
1. Go to the scope you want to create the automation in
2. Configure the event
3. Configure the action (using existing admin-created things such as secrets, webhooks)

For some of the other details, like reminding people to authorize W&B to send webhooks to a given provider, we should put that in the slack, webhook sections respectively as relevant.

Author: I removed some of the details here -- i think they were already in the Slack and Webhook pages. This is supposed to be an overview / landing page for the "Create an automation" section. This feedback is not entirely addressed yet -- I will try to finish it up tomorrow.

> Create an automation from the project or registry's **Automations** tab. At a high level, to create an automation, follow these steps:

Reviewer: can we add the relevant hyperlinks to project / registry to point the user to resources on scope?

Author: Done

> 1. If necessary, [create a W&B secret]({{< relref "/guides/core/secrets.md" >}}) for each sensitive string required by the automation, such as an access token, password, or SSH key. Secrets are defined in your team settings. Secrets are most commonly used in webhook automation.

Reviewer: ```suggestion
1. If necessary, [create a W&B secret]({{< relref "/guides/core/secrets.md" >}}) for each sensitive string required by the automation, such as an access or bearer token or an API key. Secrets are defined in your **Team Settings**. Secrets are most commonly used in webhook automations to securely pass credentials or tokens to the webhook's external service without exposing it in plain text or hard-coding it in the webhook's payload.
```

Reviewer: > you grant it access to any secrets it requires for authentication
What does mean? How does one grant authentication?

Author: This was my misunderstanding, fixed.

> 1. If necessary, [create a W&B secret]({{< relref "/guides/core/secrets.md" >}}) for each sensitive string required by the automation, such as an access token, password, or SSH key. Secrets are defined in your team settings. Secrets are most commonly used in webhook automation.

Reviewer: ```suggestion
1. If necessary, [create a W&B secret]({{< relref "/guides/core/secrets.md" >}}) for each sensitive string required by the automation, such as an access token, password, or SSH key. Secrets are defined in your **team settings**. Secrets are most commonly used in webhook automation.
```

Author: Incorporated into the other suggestion for the same line.

> 1. Configure the webhook or Slack integration to authorize W&B to post to Slack or run the webhook on your behalf. A single integration can be used by multiple automations. When you create the integration, you grant it access to any secrets it requires for authentication. Automation integrations are defined in your team settings.

Reviewer: ```suggestion
1. Configure the webhook or Slack integration to authorize W&B to post to Slack or run the webhook on your behalf. A single integration can be used by multiple automations. When you create the integration, you grant it access to any secrets it requires for authentication. Automation integrations are defined in your team settings.
```
I believe the authorization is specific to slack - some webhook providers don't require any authorization outside of using a secret. I think parts of this section should be in the Slack specific instructions.

Author: Updated, PTAL. The specific details are also in the Slack and Webhook pages, this page is just supposed to be a broad overview of the steps.

- [Learn to automatically triggers a Github Action for model evaluation and deployment](https://wandb.ai/wandb/wandb-model-cicd/reports/Model-CI-CD-with-W-B--Vmlldzo0OTcwNDQw).
- [Watch a video demonstrating automatically deploying a model to a Sagemaker endpoint](https://www.youtube.com/watch?v=s5CMj_w3DaQ).
- [Watch a video series introducing automations](https://youtube.com/playlist?list=PLD80i8An1OEGECFPgY-HPCNjXgGu-qGO6&feature=shared).
{{% /alert %}}

## Requirements
- A team admin can create and manage automations for the team's projects, as well as components of their automations, such as webhooks, secrets, or Slack connections. Refer to [Team settings]({{< relref "/guides/models/app/settings-page/team-settings/" >}}).
- To create a registry automation, you must have access to the registry. Refer to [Configure Registry access]({{< relref "/guides/core/registry/configure_registry.md#registry-roles" >}}).
- To create a Slack automation, you must have permission to post to the Slack instance and channel you select.

## Create an automation
Create an automation from the project or registry's **Automations** tab. At a high level, to create an automation, follow these steps:

> 1. If necessary, [create a W&B secret]({{< relref "/guides/core/secrets.md" >}}) for each sensitive string required by the automation, such as an access token, password, or SSH key. Secrets are defined in your **Team Settings**. Secrets are most commonly used in webhook automation.

Reviewer: ```suggestion
1. If necessary, [create a W&B secret]({{< relref "/guides/core/secrets.md" >}}) for each sensitive string required by the automation, such as an access token, password, or SSH key. Secrets are defined in your **Team Settings**. Secrets are most commonly used in webhook automations.
```

Author: This was already fixed locally before your feedback, that change is now in the PR.

> 1. Configure the webhook or Slack connection to authorize W&B to post to Slack or run the webhook on your behalf. A single connection can be used by multiple automations. These connections are defined in your **Team Settings**. 

Reviewer: ```suggestion
1. Configure the webhook or Slack notification to authorize W&B to post to Slack or run the webhook on your behalf. A single automation action (webhook or Slack notification) can be used by multiple automations. These actions are defined in your **Team Settings**. 
```

Author: Done
